src/shims/windows/thread.rs

```diff
@@ -0,0 +1,79 @@
+use std::time::{Duration, Instant};
+
+use rustc_middle::ty::layout::LayoutOf;
+use rustc_target::spec::abi::Abi;
+
+use crate::thread::Time;
+use crate::*;
+use shims::windows::handle::Handle;
+
+impl<'mir, 'tcx: 'mir> EvalContextExt<'mir, 'tcx> for crate::MiriEvalContext<'mir, 'tcx> {}
+
+#[allow(non_snake_case)]
+pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx> {
+    fn CreateThread(
+        &mut self,
+        security_op: &OpTy<'tcx, Tag>,
+        stacksize_op: &OpTy<'tcx, Tag>,
+        start_op: &OpTy<'tcx, Tag>,
+        arg_op: &OpTy<'tcx, Tag>,
+        flags_op: &OpTy<'tcx, Tag>,
+        thread_op: &OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, ThreadId> {
+        let this = self.eval_context_mut();
+
+        if !this.ptr_is_null(this.read_pointer(security_op)?)? {
```

Comment:
Please start with a section that reads all the arguments. It should be easy to see that *all* of them are checked to be valid for whatever type we want them at.